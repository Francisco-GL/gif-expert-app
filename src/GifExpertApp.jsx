import { useState } from 'react';
import { AddCategory, GifGrid } from './components';
// import { AddCategory } from './components/AddCategory';
// import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setcategories] = useState([ 'One Punch' ]);

    const onAddCategory = ( newcategory ) => {

        if(categories.includes(newcategory)) return;

        setcategories( [ newcategory, ...categories] );
        // setcategories( cat => [ ...cat, 'Valorant'] );
    }

  return (
    <>
        <h1>GifExpertApp</h1> 

        <AddCategory 
            // setCategories={ setcategories} 
            onNewCategory={ value => onAddCategory(value) }
        />

        { 
            categories.map( (category) => (
                <GifGrid key={ category } category={ category }/>
            )) 
        }
    </>
  )
}
