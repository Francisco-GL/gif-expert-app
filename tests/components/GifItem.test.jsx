import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Pruebas en <GifItem/>', () => {

    const title = 'One-Puch';
    const url = 'http://localhost/holaMundo.jpg';
  
    test('debe de hacer match con el snapshot ', () => {
      const {container } = render(<GifItem title={ title } url={ url }/>);
      
      expect( container ).toMatchSnapshot();
    });

    test('dsebe de mostrar la imagen con el url y el ALT indicado', () => {
      

        render(<GifItem title={ title } url={ url }/>);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );


    });

    test('debe de mostrar el titulo en el coponente', () => {
      render( <GifItem title={ title } url={ url }/>);
      expect( screen.getByText(title) ).toBeTruthy();
    })
    
    
    
});
