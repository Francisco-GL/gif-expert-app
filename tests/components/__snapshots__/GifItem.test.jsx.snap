// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One-Puch"
      src="http://localhost/holaMundo.jpg"
    />
    <p>
      One-Puch
    </p>
  </div>
</div>
`;
